src/web/transactions.rs

```diff
@@ -23,62 +25,45 @@ pub enum TransactionDTO {
     Rejected(RejectedTransactionDTO),
 }
 
-impl TryFrom<TransactionValue> for TransactionDTO {
+impl TryFrom<TransactionQueryResult> for TransactionDTO {
```

Comment:
`TransactionQueryResult` is a structure with block hash and transaction value. As I see below, you only use the internal `TransactionValue` to build `TransactionDTO` from it.  Therefore, it is not necessary to transform a whole `TransactionQueryResult` if only its `TransactionValue` is needed.

Reply:
As we would need ```block_hash``` and it is in the ```TransactionQueryResult```, not in the ```TransactionValue```. Therefore, we need both the ```block_hash``` and the ```TransactionValue```, so will indeed have to convert the entire ```TransactionQueryResult``` to build the ```TransactionDTO```.

Reply:
Indeed. I haven't noticed that `block_hash` is used as well.